### PR TITLE
Add rubygem

### PR DESCRIPTION
**Pull request by tsmartt**
I've tested this works locally and will integrate this gem into creators as soon as merged.

### CODE REVIEW

File: wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        rescue RuntimeError
+          raise "Error processing the image."
+        end
+        # Get a pointer to the result
+        memory = wasm_instance.exports.memory.uint8_view output_pointer
+
+        # Only take the buffer that we told the rust function we needed. The resize function
+        # makes a smaller image than the buffer we said, and then pads out the rest so we have to
+        # go hunting for the bytes which represent the JPEG image. In hex, JPEG images start with
+        # FFD8 and FFD9, so we can convert to hex and find the bounds of the image, then write to file
```
Comment (evq) [inline wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb:65]: we can drop this comment about hunting for magic bytes - right?

File: wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        wasm_instance.exports.deallocate.call(input_pointer, image_length)
+        wasm_instance.exports.deallocate.call(output_pointer, bytes.length)
+
+        # The bytes passed back to us are ASCII-encoded, i.e. 8bit bytes. Interpret them as so,
+        # and THEN convert to hex to search for the image bytes
+
+        # The first 4 bytes are a header until the image. The actual image probably ends well before
+        # the whole buffer, but we keep the junk data on the end to make all the images the same size
+        # for privacy concerns.
+        bytes[4..].pack("C*")
```
Comment (evq) [inline wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb:78]: non-blocking but it's slightly strange that our gem wrapper removes the prefix bytes but the python version doesn't

**Review by evq — DISMISSED** [review]

File: wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb (hunk 1 above)
Comment (tsmartt) [reply wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb]: Done.

File: wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb (hunk 2 above)
Comment (tsmartt) [reply wasm-thumbnail-rb/lib/wasm/thumbnail/rb.rb:79]: Done.

**Review by evq — APPROVED** [review]

**Pull request merged**